agimus_controller/agimus_controller/robot_model_factory/robot_model_factory.py

```diff
@@ -0,0 +1,144 @@
+from typing import Union
+
+try:
+    from typing import Self
+except ImportError:
+    from typing_extensions import Self
+from dataclasses import dataclass
+from copy import deepcopy
+import pinocchio as pin
+import numpy as np
+from pathlib import Path
+from .obstacle_params_parser import ObstacleParamsParser
+
+
+@dataclass
+class RobotModelParameters:
+    q0_name = str()
+    free_flyer = False
+    locked_joint_names = []
+    urdf = Path()
+    srdf = Path()
+    collision_as_capsule = False
+    self_collision = False
+
+
+class RobotModelFactory:
+    """Complete model of the robot."""
+
+    _model = pin.Model()
+    """ Complete model of the robot with visualization meshes. """
+    _cmodel = pin.GeometryModel()
+    """ Complete model of the robot with collision meshes. """
+    _vmodel = pin.GeometryModel()
+    """ Reduced model of the robot. """
+    _rmodel = pin.Model()
+    """ Reduced model of the robot with visualization meshes. """
+    _rcmodel = pin.GeometryModel()
+    """ Reduced model of the robot with collision meshes. """
+    _rvmodel = pin.GeometryModel()
+    """ Default configuration q0. """
```

Comment:
wrong comment, should be 

```suggestion
    """ Reduced visual model of the robot (with collision meshes).  """
```